### ISSUE TITLE
Spam protection methods

### ISSUE THREAD

**Issue by p2prider** [p2prider/p2prider#8]
We should have some simple ways to vet riders and drivers to prevent spam. We should also be mindful of the privacy of all users, whether drivers or riders.

1. Mobile OTP
    - This is the simplest option and can be implemented cheaply as well with twilio APIs.
    - Good UX too as most users would be using it on their phone.
    - Not the best option for privacy. Users may not be happy to share their phone number to an app not incorporated/backed by a known brand.
2. UPI id
    - We can send a charge of 1 rupee to the UPI id entered by the user to verify them
    - Auto drivers will register with UPI ID anyway (the app will show the driver's UPI ID to the rider so the rider can pay quickly), so this will be seamless identification and verification for them
    - Most banks and UPI apps allow creating virtual UPI ids which users can use for added privacy
3. Invite based
    - An existing verified user can generate an invite code and send it to the new driver/rider which they can use to signup. The system will keep a graph of who invited whom.
    - If/when spam is detected, the offending user will be suspended.
    - The user who invited the spammer can be held accountable, either by suspension or by preventing them from inviting anyone else.

Thoughts?

**Comment by amoghamblihalli** [comment]
initially, invite-based seems fine (might be too restricted though). however OTP-based is the usual standard and much more accessible to both user and driver.

**Comment by vvkishere** [comment]
An area that I find to be challenging would be complaints redressal. 

How would a customer get an issue sorted out within this system? For instance if there is a complaint about a passenger being threatened who do they go to and get this sorted out and ensure the driver doesn't get assigned more customers? How do you make sure the complaints are genuine? 

We need a way to make this system smooth without the dependency on a central system?

**Comment by p2prider** [comment]
@vvkishere I don't think we can do much beyond providing a ratings/reviews system. Providing customer support would be very expensive. We should provide safety features such as easy way to contact police, sharing location with friends/family etc. It's a community project, even the users are part of the community. The developers are also gonna be the users. Naively optimistic, perhaps, but I do think this system can work without having a customer support.

**Comment by ashwingattani** [comment]
> @vvkishere I don't think we can do much beyond providing a ratings/reviews system. Providing customer support would be very expensive. We should provide safety features such as easy way to contact police, sharing location with friends/family etc. It's a community project, even the users are part of the community. The developers are also gonna be the users. Naively optimistic, perhaps, but I do think this system can work without having a customer support.

more on this is being discussed [here](https://github.com/p2prider/p2prider/issues/13)
